getMongoData/getMongoData.js

```diff
@@ -604,36 +655,45 @@ if (typeof db.printSecondaryReplicationInfo === 'function') {
 }
 
 var _total_collection_ct = 0;
-var _output = [];
 var _tag = ObjectId();
 if (! _printJSON) {
     print("================================");
     print("MongoDB Config and Schema Report");
     print("getMongoData.js version " + _version);
     print("================================");
+} else {
+    // Start the JSON array.
+    print("[\n");
 }
 
 var _host = hostname();
-
+var _error = null;
 try {
     printServerInfo();
     var isMongoS = printShardOrReplicaSetInfo();
     printUserAuthInfo();
     printDataInfo(isMongoS);
 } catch(e) {
-    // To ensure that the operator knows there was an error, print the error
-    // even when outputting JSON to make it invalid JSON.
-    print('\nERROR: '+e.message);
-
+    _error = e.message;
     if (e.name === 'MaxCollectionsExceededException') {
-      // Prefix the "subsection" fields with "INCOMPLETE_" to make
-      // it clear that the database and collection info are likely to be
-      // incomplete.
-      updateDataInfoAsIncomplete(isMongoS);
+        printInfo("incomplete_databases_and_collections_info", function(){ return e.message; });
```

Comment:
Besides the `INCOMPLETE` output behaviour, does this patch change any other output with default settings, both in JSON and non-JSON mode?

Reply:
No, except the INCOMPLETE behaviour I have strived to preserve the default behaviour.

Reply:
Well, there's a slight difference where before the error was printed before the JSON, now this is not possible and the error is printed after the JSON.

Reply:
Can you add a comment on top of the `var _version` definition with a summary of possibly breaking changes from 3.1.0? That will help understand if there's any downstream impact, when we start upstreaming this work.